src/app.rs

```diff
@@ -0,0 +1,138 @@
+use crate::cli::{Cli, Command};
+use crate::cloud_client::CloudClient;
+use crate::errors::PARSE_COMMAND_ERROR;
+use crate::tui::WorkMode;
+use crate::utilities::files::get_path_entries;
+use std::path::PathBuf;
+use tracing::debug;
+
+pub struct WorkspaceData {
+    pub local_path: PathBuf,
+    pub local_entries: Vec<String>,
+    pub cloud_path: PathBuf,
+    pub cloud_entries: Vec<String>,
+}
+
+impl Default for WorkspaceData {
+    fn default() -> Self {
+        Self {
+            local_path: std::env::current_dir().unwrap_or_default(),
+            local_entries: vec![],
+            cloud_path: PathBuf::from("//"),
+            cloud_entries: vec![],
+        }
+    }
+}
+
+pub struct App<C: CloudClient> {
+    pub input_command: String,
```

Comment:
Can this `String` commend be replaced with a typed one (e.g. some `enum`)?

Reply:
No, this is string that was read from input layout in tui. This string will be parsed in `App::submit_command`

Reply:
@BigglesworthCat please, answer this one (I'm not telling you to fix it. just answer)

Reply:
No, it can't be replaced due to way user write those commands